burnman/eos/vinet.py

```diff
@@ -0,0 +1,127 @@
+# This file is part of BurnMan - a thermoelastic and thermodynamic toolkit for the Earth and Planetary Sciences
+# Copyright (C) 2012 - 2015 by the BurnMan team, released under the GNU GPL v2 or later.
+
+
+import scipy.optimize as opt
+from . import equation_of_state as eos
+import warnings
+from math import exp
+
+def bulk_modulus(volume, params):
+    """
+    compute the bulk modulus as per the third order
+    Vinet equation of state.  Returns bulk
+    modulus in the same units as the reference bulk
+    modulus.  Pressure must be in :math:`[Pa]`.
+    """
+
+    x = volume/params['V_0']
+    eta = (3./2.)*(params['Kprime_0']-1.)
+
+    K = (params['K_0']*pow(x,-2./3.))*(1+((eta*pow(x,1./3.)+1.)*(1.-pow(x,1./3.))))*exp(eta*(1.-pow(x,1./3.)))
+    return K
+
+def vinet(x, params):
+    """
+    equation for the third order Vinet equation of state, returns
+    pressure in the same units that are supplied for the reference bulk
+    modulus (params['K_0'])
+    """
+    eta = (3./2.)*(params['Kprime_0']-1.)
+    return 3.*params['K_0'] * (pow(x, -2./3.))*(1.-(pow(x,1./3.))) \
+    * exp(eta*(1-pow(x,1/3.)))
+
+def volume(pressure, params):
+    """
+    Get the Vinet volume at a reference temperature for a given
+    pressure :math:`[Pa]`. Returns molar volume in :math:`[m^3]`
+    """
+
+    func = lambda x: Vinet(x/params['V_0'], params) - pressure
```

Comment:
Please run "python tests.py" on your machine. You will see an error triggered here (it should be lowercase "vinet").